packages/@lwc/template-compiler/src/parser/parser.ts

```diff
@@ -121,6 +134,25 @@ export default class ParserCtx {
         });
     }
 
+    warnOnIRNode(
+        errorInfo: LWCErrorInfo,
+        irNode: IRNode | IRBaseAttribute,
+        messageArgs?: any[]
+    ): void {
+        this.warnAtLocation(errorInfo, messageArgs, irNode.location);
+    }
+
+    warnAtLocation(errorInfo: LWCErrorInfo, messageArgs?: any[], location?: parse5.Location): void {
```

Comment:
Maybe for a different PR. It is strange for a method called `warnAtLocation` to accept an optional `location` argument. The same observation is also valid for `throwAtLocation`. Those methods should either be renamed or should require the location.

Reply:
I'll add this to the next PR.